extensions/wikia/Chat2/js/emoticons.js

```diff
@@ -35,49 +35,47 @@ WikiaEmoticons.doReplacements = function (text, emoticonMapping) {
 	$().log('Processing any emoticons... ');
 
 	var imgUrlsByRegexString = emoticonMapping.getImgUrlsByRegexString(),
-		regexString,
-		imgSrc,
-		numIters,
 		origText,
 		regex,
-		glyphUsed,
-		buildTagFunc;
-
-	for (regexString in imgUrlsByRegexString){
-		// Empty string for emote icons crash Chat so ignore them
-		if (regexString === '') {
-			continue;
-		}
+		buildTagFunc,
+		maxEmoticons = 5,
+		combinedRegex = Object.keys(imgUrlsByRegexString)
+			.map(function (key) {
+					return key.replace(/[-[\]{}()*+?.,\\^$|#\s]/g, '\\$&');
+				}
+			).join('|');
 
-		imgSrc = imgUrlsByRegexString[regexString];
-		imgSrc = imgSrc.replace(/"/g, '%22'); // prevent any HTML-injection
+	//return early if none match
+	if (!text.match(combinedRegex)) {
+		return text;
+	}
 
-		buildTagFunc = WikiaEmoticons.buildTagGenerator(imgSrc);
+	// Fix > and <
+	combinedRegex = combinedRegex.replace(/>/g, '&gt;');
+	combinedRegex = combinedRegex.replace(/</g, '&lt;');
 
-		// Fix > and <
-		regexString = regexString.replace(/>/g, '&gt;');
-		regexString = regexString.replace(/</g, '&lt;');
+	buildTagFunc = WikiaEmoticons.buildTagGenerator(imgUrlsByRegexString);
 
-		// Build the regex for the character (make it ignore the match if there is a "/" immediately
-		// after the emoticon. That creates all kinds of problems with URLs).
-		numIters = 0;
+	do {
 		origText = text;
-
-		do {
-			// NOTE: \s does not work for whitespace here for some reason.
-			regex = new RegExp('(^| )(' + regexString + ')([^/]|$)', 'gi');
-			glyphUsed = text.replace(regex, '$2');
-			glyphUsed = glyphUsed.replace(/"/g, '&quot;'); // prevent any HTML-injection
-			text = text.replace(regex, buildTagFunc);
-		} while ((origText !== text) && (numIters++ < 5));
-	}
+		regex = new RegExp('(^|\\s)(' + combinedRegex + ')([^/]|$)', 'i');
```

Comment:
As discussed offline, either:

* remove the loop (i.e. emoticons per message limit) and add `g` flag to the regexp
* create RegExp instance outside of do / while loop